#22622

cristianhosu opened the pull request:
Please validate the layout and the implementation.
Need help making the scroll work from the mouse wheel inside the .debug-hover-widget

msftclas commented:
@cristianhosu,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by Microsoft. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

isidorn commented:
@cristianhosu thank you for your PR. However we are currently in the endgame, which means we are wrapping up our March milestone. Due to that I will check this PR and provide feedback next week.

cristianhosu commented:
@isidorn Hello, can you please check?
Thank you

isidorn commented:
@cristianhosu yes I plan to look into this today / tomorrow. Thanks for the reminder

isidorn commented:
@cristianhosu great work!
Why did you add the z-index - that does not seem needed? Everything else looks good

@joaomoreno how does the suggest widget steal the scrolling from the editor?

cristianhosu commented:
@isidorn Thank you
I removed the z-index. I added it as a test to try to put the debug panel on top of the underlying panel to make it scrollable when inside it.

isidorn commented:
@cristianhosu sorry for the slow response, last week I was on vacation.
I still see the z-index, are you sure you pushed the commit on top of this pr.

isidorn commented:
Pushing to May since we are now in endgame, will merge it in next week.
Sorry for the delay, there was some bad timing regarding my vacation

cristianhosu commented:
@isidorn I've pushed to the correct branch as far as i can tell, but AppVeyor and Travis CI have not completed. maybe this is the reason why it's not visible?

isidorn commented:
@cristianhosu now it is good. I will merge it in and look into the scrolling issue. Thanks

isidorn commented:
I have looked into the scrolling issue and to make this proper we need to use the [`DomScrollableElement`](https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/ui/scrollbar/scrollableElement.ts#L370) class. This will also give us consistent look and feel across different OS.
I am looking into this...

isidorn commented:
I am running out of cycles for this milestone and will not have time to look into using [`DomScollableElement`](https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/ui/scrollbar/scrollableElement.ts#L370) class. @cristianhosu are you intrested into looking into this? Using this class will give us consistent scroll bar design across vscode. Nowhere else do we use the native scroll bar and that is the main reason why this PR can not be merged in right now.

cristianhosu commented:
Yup, I'll look into it :)

On Mon, May 22, 2017 at 12:00 PM, Isidor Nikolic <notifications@github.com>
wrote:

> I am running out of cycles for this milestone and will not have time to
> look into using DomScollableElement
> <https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/ui/scrollbar/scrollableElement.ts#L370>
> class. @cristianhosu <https://github.com/cristianhosu> are you intrested
> into looking into this? Using this class will give us consistent scroll bar
> design across vscode. Nowhere else do we use the native scroll bar and that
> is the main reason why this PR can not be merged in right now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/vscode/pull/23211#issuecomment-303041348>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQ3E7mY31UCmWvm6EGrXFucEvHujDIHPks5r8U6tgaJpZM4MpHZ->
> .
>



-- 
Cristian Hosu
+40740.058.711

isidorn commented:
@cristianhosu great, thanks!

cristianhosu commented:
So, i've investigated and i've also made some changes but i can't seem to
run VS Code OSS Dev with all the expansions :)
I have my test app with some node.js and I cannot start that app from
within the OSS Dev version.
Can you please help with this?

I haven't changed anything in the launch.json of the vscode project, but i
think i should, like maybe a cmd line parameter?

On Tue, May 23, 2017 at 10:34 AM, Isidor Nikolic <notifications@github.com>
wrote:

> @cristianhosu <https://github.com/cristianhosu> great, thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/vscode/pull/23211#issuecomment-303314718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQ3E7rM8u11rjKeZj7NId6C1KkFxJpcOks5r8ovzgaJpZM4MpHZ->
> .
>



-- 
Cristian Hosu
+40740.058.711

isidorn commented:
@cristianhosu thanks for looking into this further.
Yes, this is a common issue people hit and we should improve this. You can find details on how to fix this in my comment here

One solution https://github.com/Microsoft/vscode/issues/20623#issuecomment-279969736
Second solution is here https://github.com/Microsoft/vscode/issues/23831#issuecomment-291233092

isidorn commented:
@cristianhosu let me know if there is anything else I can help with

cristianhosu commented:
@isidorn I have finished. I've finally made time to investigate how to properly implement a DomScrollableElement and it worked.

isidorn commented:
@cristianhosu great! Feel free to push those commits once they are ready so I can review them.

cristianhosu commented:
@isidorn I have pushed. Sorry about that... i keep forgetting to do so, being used to TFS :)

isidorn commented:
@cristianhosu absolutetly great work! Thanks a lot!
Just one more tiny thing: can you please remove the `ScrollbarVisibility` auto options passed on line 66, those are the defaults and we should not specify them. Once you do that you will notice that one import you added is not necessery.

isidorn commented:
Great work

cristianhosu commented:
@isidorn - thank you :)
